libs/bn-common/src/components/BoatnetList.vue

```diff
@@ -40,39 +41,53 @@ export default createComponent({
   },
 
   setup(props, context) {
-    const lists = props.list ? props.list : [''];
+    const modelName: string = props.config ? props.config.modelName : '';
+    const maxListSize: number = props.config ? props.config.maxItems : 0;
+    const minListSize: number = props.config ? props.config.minItems : 1;
+
+    const lists: string[] = initList();
     const deleteMessage = ref('');
     const showDeleteDialog = ref(false);
     let deleteIndex: number = 0;
 
-    const modelName: string = props.config ? props.config.modelName : '';
-    const maxListSize: number = props.config ? props.config.maxItems : 0;
+    function initList() {
+      const items: any[] = props.list ? props.list : [];
+      if (items.length === 0) {
+        for (let i = 0; i < minListSize; i++) {
+          items.push('');
+        }
+      }
+      return items;
+    }
 
-    const save = () => {
+    function save(index: number) {
       context.emit('update:list', lists);
       context.emit('save');
-    };
+    }
 
-    const confirmDelete = (index: number) => {
+    function confirmDelete(index: number) {
       context.root.$store.dispatch('keyboard/setKeyboard', false);
       showDeleteDialog.value = true;
-      deleteMessage.value = 'Are you sure you want to delete this entry?';
+      const displayName = props.config ? props.config.displayName : '';
+      deleteMessage.value = 'Are you sure you want to delete '
+                          + displayName + ': '
+                          + lists[index] + '?';
       deleteIndex = index;
-    };
+    }
 
-    const onDelete = () => {
+    function onDelete() {
       lists.splice(deleteIndex, 1);
       context.emit('save');
-    };
+    }
 
-    const add = () => {
+    function add() {
       context.root.$store.dispatch('keyboard/setKeyboard', false);
       if (lists.length < maxListSize) {
-        lists.splice(0, 0, '');
```

Comment:
I suspect the reason for this ```splice(0,0,'');``` is that it keeps the list reactive, where ```push('')``` does not. 

Review this: https://vuejs.org/v2/guide/list.html#Caveats

Not sure if this matters, but good to be aware of.